lib/backend/test/put_batch.go

```diff
@@ -0,0 +1,194 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/backend"
+)
+
+const (
+	watchInitTimeout  = 10 * time.Second
+	watchEventTimeout = 3 * time.Second
+)
+
+func runPutBatch(t *testing.T, newBackend Constructor) {
+	t.Helper()
+
+	ctx := context.Background()
+	bk, _, err := newBackend()
+	require.NoError(t, err)
+	t.Cleanup(func() { _ = bk.Close() })
+
+	batcher, ok := bk.(backend.BatchPutter)
+	if !ok {
+		t.Skip("backend does not implement PutBatch; skipping PutBatch suite")
+	}
+
+	prefix := MakePrefix()
+	rangeStart := prefix("")
+	rangeEnd := backend.RangeEnd(prefix(""))
+
+	itemEqual := func(a, b backend.Item) bool {
+		return a.Key.String() == b.Key.String() &&
+			a.Revision == b.Revision &&
+			string(a.Value) == string(b.Value) &&
+			a.Expires.Equal(b.Expires)
+	}
+
+	assertItemsEqual := func(t *testing.T, want, got []backend.Item) {
+		t.Helper()
+		require.Len(t, want, len(got))
+		for i := range want {
+			require.True(t, itemEqual(want[i], got[i]))
+		}
+	}
+
+	buildWant := func(items []backend.Item, rev []string) []backend.Item {
+		out := make([]backend.Item, 0, len(items))
+		for i, it := range items {
+			out = append(out, backend.Item{
+				Key:      it.Key,
+				Value:    it.Value,
+				Revision: rev[i],
+				Expires:  it.Expires,
+			})
+		}
+		return out
+	}
+
+	newTestItems := func() []backend.Item {
+		return []backend.Item{
+			{Key: prefix("a"), Value: []byte("A"), Expires: time.Now().Add(1 * time.Hour)},
+			{Key: prefix("b"), Value: []byte("B")},
+			{Key: prefix("c"), Value: []byte("C"), Expires: time.Now().Add(2 * time.Hour)},
+		}
+	}
+	t.Run("put batch items should be propagated in event stream", func(t *testing.T) {
+		w, err := bk.NewWatcher(t.Context(), backend.Watch{})
```

Comment:
```suggestion
		w, err := bk.NewWatcher(ctx, backend.Watch{})
```

Reply:
In needs to be sub test context no global context.